pkgs/tools/misc/opentelemetry-collector/contrib.nix

```diff
@@ -2,28 +2,33 @@
 , fetchFromGitHub
 , lib
 , stdenv
+, systemdMinimal
 }:
 
 buildGoModule rec {
   pname = "opentelemetry-collector-contrib";
-  version = "0.66.0";
+  version = "0.76.1";
 
   src = fetchFromGitHub {
     owner = "open-telemetry";
     repo = "opentelemetry-collector-contrib";
     rev = "v${version}";
-    sha256 = "sha256-FT5AoqCHNf2sdKyejALOsL/zHrrxP7vdntagR9vA00I=";
+    sha256 = "sha256-Aeiq9IJReUxJUpeq5mSReK5foC5aY4fMSZli0ZUjYPc=";
   };
   # proxy vendor to avoid hash missmatches between linux and macOS
   proxyVendor = true;
-  vendorSha256 = "sha256-65bfTCMRJ8iL5ABGPqvkayw4zSn4KkCriEkWYa0Pe68=";
+  vendorSha256 = "sha256-vLbx/qmSZuteuvChnyA/wcEcIjU3zWkxSjfk8VBdgU4=";
 
-  subPackages = [ "cmd/otelcontribcol" ];
+  # there is a nested go.mod
+  sourceRoot = "source/cmd/otelcontribcol";
 
   # CGO_ENABLED=0 required for mac - "error: 'TARGET_OS_MAC' is not defined, evaluates to 0"
   # https://github.com/shirou/gopsutil/issues/976
   CGO_ENABLED = if stdenv.isLinux then 1 else 0;
 
+  # journalctl is required for the journald receiver
```

Comment:
is it that journalctl is required at build time, systemd headers are required at build time, or journalctl is required at runtime?

Just scanning over [`reciever/journaldreceiver`'s README](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/journaldreceiver/README.md) it looks like a runtime dependency so this won't help here.

We can add to `propogatedBuildInputs` to ensure `journalctl` exists but unless it prevents opentelemetry-collector-contrib from working I'd much rather pick up `journalctl` from the PATH than bring in the whole of `systemdMinimal` (even if it is meant to be minimal, on systems that don't have systemd...)

or we can do a `withSystemd ? false` option people can turn true if they really want

Reply:
I think your feedback is right, I've edited my commit and squashed/rebase in c151963, let me know if that looks good to you.